main.py

```diff
@@ -1,99 +1,136 @@
 import os
 import argparse
 import numpy as np
+from tqdm import tqdm
+import typing
+
 
 import torch
 import torch.nn as nn
 import torch.nn.functional as F
-from torch.utils.data import Dataset, DataLoader,random_split
+from torch.utils.data import Dataset, DataLoader, random_split
 from torch.optim import AdamW
 from torch.optim.lr_scheduler import ExponentialLR
 
 import pytorch_lightning as pl
-from pytorch_lightning import LightningDataModule, LightningModule, Trainer, seed_everything
+from pytorch_lightning import (
+    LightningDataModule,
+    LightningModule,
+    Trainer,
+    seed_everything,
+)
 from pytorch_lightning.callbacks.progress import TQDMProgressBar
 from pytorch_lightning.callbacks import ModelCheckpoint
 
 from data import Microstructures
-#from unet import UNet
+
+# from unet import UNet
 from unet_attention import UNet
-from sampler import DDIMSampler
+
+# from sampler import DDIMSampler
+from diffusers import DDPMScheduler, DDIMScheduler
 
 import warnings
-warnings.filterwarnings('ignore')
+
+
+warnings.filterwarnings("ignore")
 
 
 def main():
-    
+
     global args
-    
-    if args.n_epochs < 5*args.sample_interval:
-        args.sample_interval = args.n_epochs//6
+
+    if args.n_epochs < 5 * args.sample_interval:
```

Comment:
add a comment line or two here; it's not obvious what `sample_interval` is or why it should/shouldn't be related to epoch count like this